commands/other/event.js

```diff
@@ -0,0 +1,93 @@
+const { Command } = require('discord.js-commando');
+const { MessageEmbed } = require('discord.js');
+const { default: Axios } = require('axios');
+require('dotenv').config();
+const APIKey = process.env.IGDB_API_KEY;
+const axios = require('axios');
+
+module.exports = class eventCommand extends Command {
+  constructor(client) {
+    super(client, {
+      name: 'event',
+      aliases: ['trourney', 'party'],
+      group: 'other',
+      memberName: 'event',
+      description: 'Announces a server event',
+      examples: ['!event "game" "number of players" "day" "time"'],
+      guildOnly: true,
+      clientPermissions: ['SPEAK', 'CONNECT'],
+      args: [
+        {
+          key: 'queryGame',
+          prompt: 'What game would you like to play?',
+          type: 'string',
+        },
+        {
+          key: 'queryPlayerNum',
+          prompt: 'How many players?',
+          type: 'integer',
+        },
+        {
+          key: 'queryEventDay',
+          prompt: 'What Day?',
+          type: 'string',
+        },
+        {
+          key: 'queryEventTime',
+          prompt: 'What Time?',
+          type: 'string',
+        },
+      ],
+    });
+  }
+
+  hasPermission(message) {
+    const approvedRoles = ['🛡 Division Commander', '⚔️ Commander'];
+    const title = message.member.roles.highest.name;
+    if (approvedRoles.includes(title)) return true;
+    return 'Only Division Commanders and above may create events.';
+  }
+
+  async run(
+    message,
+    { queryGame, queryPlayerNum, queryEventDay, queryEventTime }
+  ) {
+    const embed = new MessageEmbed();
+    const eventTime = queryEventTime;
+    const eventDay = queryEventDay;
+    const eventGame = queryGame;
```

Comment:
Why make copies of these variables? just use the ones you create in the run

Reply:
I wanted to convert time and dates to actually time and date format I can strip them out for now if you want. Date will need to be a date format once the firebase is setup so we can use it to create temporary roles for events and have them automatically expire after the event date.